CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+set(CHPL_CMAKE_TARGET_MEM CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_JEMALLOC CACHE STRING "Description")
+set(CHPL_CMAKE_MAKE CACHE STRING "Description")
+set(CHPL_CMAKE_GMP CACHE STRING "Description")
+set(CHPL_CMAKE_GMP_IS_OVERRIDDEN CACHE STRING "Description")
+set(CHPL_CMAKE_RE2_IS_OVERRIDDEN CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_SUPPORT CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_CONFIG CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_VERSION CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_CLANG_C CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_CLANG_CXX CACHE STRING "Description")
+set(CHPL_CMAKE_SANITIZE CACHE STRING "Description")
+set(CHPL_CMAKE_RUNTIME_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_LAUNCHER_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_COMPILER_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_BIN_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_BIN_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_SYS_MODULES_SUBDIR CACHE STRING "Description")
+set(CHPL_CMAKE_LLVM_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_GASNET_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_GMP_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_HWLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_HOST_JEMALLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_TARGET_JEMALLOC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_LIBFABRIC_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_LIBUNWIND_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_QTHREAD_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_RE2_UNIQ_CFG_PATH CACHE STRING "Description")
+set(CHPL_CMAKE_PE_CHPL_PKGCONFIG_LIBS CACHE STRING "Description")
+
+set(PRINTCHPLENV_COMMAND printchplenv --compiler --internal --cmake)
+# message(STATUS "running ${PRINTCHPLENV_COMMAND} from ${CMAKE_CURRENT_SOURCE_DIR}/util")
+
+find_package(Python)
+
+execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/util/${PRINTCHPLENV_COMMAND}
+                WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/util
+                TIMEOUT 5
+                RESULT_VARIABLE CHPLENV_RETURNCODE
+                OUTPUT_VARIABLE CHPLENV_OUTPUT
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+                )
+
+# string(REGEX REPLACE "\n$" "; " CHPLENV_OUTPUT "${CHPLENV_OUTPUT}")
+# This is a workaround for CMake not having any in-built way to
+# split a string into a list.  It's not perfect, but it works for this purpose
+STRING(REGEX REPLACE ";" "\\\\;" CHPLENV_OUTPUT "${CHPLENV_OUTPUT}")
+STRING(REGEX REPLACE "\n" ";" CHPLENV_OUTPUT "${CHPLENV_OUTPUT}")
+# message(DEBUG "printchplenv returned ${CHPLENV_RETURNCODE}")
+# message(DEBUG "got the following from printchplenv: ${CHPLENV_OUTPUT}")
+# set all the chplenv values from printchplenv
+foreach (CHPLENV_LINE IN LISTS CHPLENV_OUTPUT)
```

Comment:
Why was it necessary to modify printchplenv to print out cmake code if the cmake code here is going to parse it anyway? Seems like it could use `--simple` instead.

Reply:
Originally I had `printchplenv` writing the complete CMake command, e.g., `set(CHPL_VAR VALUE ...)`, but then I couldn't figure out how to execute those lines, so went back to forming the command in CMake. I left it incase we come up with a better format, but I suppose it could be removed if we can rely on something like `--simple`

Reply:
Actually I was just thinking that having "cmake mode" for printchplenv is good because we can output the things that are really lists in the Python (e.g. arguments for compilation commands) as cmake lists (which, AFAIK, actually just means separating the elements with `;`).